File: programmers/level2/기능개발.py
def solution(progresses, speeds):
    answer = []
    cnt = 0
    idx = 0
    while True:
        for i in range(idx, len(progresses)):
            progresses[i] += speeds[i]

        if progresses[idx] >= 100:
            for j in range(idx, len(progresses)):
                if progresses[j] >= 100:
                    cnt += 1
                    idx += 1
                else:
                    break
        if cnt != 0:
            answer.append(cnt)
        cnt = 0

        if idx == len(progresses):
            break

    return answer

progresses = [93, 30, 55]
speeds = [1, 30, 5]
print(solution(progresses, speeds))

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] answer = {};
        ArrayList<Integer> arr = new ArrayList<>();
        Deque<Integer> dq = new ArrayDeque<>();
        
        for(int i=0; i<progresses.length; i++) {
            int day = 0;
            while(progresses[i]<100) {
                progresses[i] += speeds[i];
                day++;
            }
            dq.offer(day);
        }
        
        int temp = dq.poll();
        int cnt = 1;
        while(!dq.isEmpty()) {
            int num = dq.poll();
            if(num > temp) {
                temp = num;
                arr.add(cnt);
                cnt = 1;
            } else cnt++;
        }
        arr.add(cnt);
        
        answer = arr.stream().mapToInt(i->i).toArray();
        return answer;
    }
}